### ISSUE TITLE
list throws

### ISSUE THREAD

**Issue by emmanueltouzery** [funkia/list#68]
I got this stack when playing with list in my fuzzer (2.0.14) =>

     TypeError: Cannot read property 'sizes' of undefined
      at sliceRight (node_modules/list/dist/index.js:1220:49)
      at sliceRight (node_modules/list/dist/index.js:1227:21)
      at slice (node_modules/list/dist/index.js:1347:28)
      at Object.take (node_modules/list/dist/index.js:1360:12)
      at Vector.take (dist/src/Vector.js:613:29)
      at Array.<anonymous> (dist/tests/Seq.js:362:62)
      at Context.<anonymous> (dist/tests/Seq.js:384:28)

presumably that should never happen. I have no reproduction for now.

**Comment by emmanueltouzery** [comment]
I have a reproduction for that stack, it's a bit long though:

<details>
[ 'map *2',
  'reverse',
  'reverse',
  'map *2',
  'reverse',
  'reverse',
  'reverse',
  'drop 232',
  'prepend 25',
  'map *2',
  'take 166',
  'map *2',
  'filter >=162',
  'reverse',
  'map *2',
  'append 139',
  'reverse',
  'append 177',
  'reverse',
  'take 215',
  'drop 46',
  'prepend 116',
  'map *2',
  'append 216',
  'append 149',
  'append 32',
  'take 11',
  'reverse',
  'append 28',
  'map *2',
  'reverse',
  'append 178',
  'append 98',
  'map *2',
  'drop 21',
  'map *2',
  'append 153',
  'prepend 213',
  'append 82',
  'map *2',
  'append 14',
  'reverse',
  'append 244',
  'reverse',
  'append 27',
  'append 169',
  'drop 152',
  'take 149',
  'drop 26' ]
</details>

**Comment by emmanueltouzery** [comment]
that repro is for the exact same stack I reported =>

<details>
[ 'append 105',
  'map *2',
  'append 228',
  'map *2',
  'map *2',
  'append 3',
  'reverse',
  'reverse',
  'map *2',
  'take 129',
  'append 135',
  'prepend 79',
  'reverse',
  'prepend 190',
  'append 143',
  'reverse',
  'map *2',
  'append 193',
  'append 115',
  'reverse',
  'append 136',
  'prepend 131',
  'reverse',
  'append 231',
  'drop 180',
  'take 95' ]

</details>

**Comment by emmanueltouzery** [comment]
BTW I realized that I'm actually reproducing this in real-world code, the code which generates apidoc for the prelude.ts library.

Fetching the prelude.ts master source, running `npm install`, `tsc`, then `npm run docgen` should reproduce the issue.

```
/home/emmanuel/home/prelude.ts/node_modules/list/dist/index.js:1220
    var _a = getPath(index, offset, depth, node.sizes),
                                                ^

TypeError: Cannot read property 'sizes' of undefined
    at sliceRight (/home/emmanuel/home/prelude.ts/node_modules/list/dist/index.js:1220:49)
    at sliceRight (/home/emmanuel/home/prelude.ts/node_modules/list/dist/index.js:1227:21)
    at slice (/home/emmanuel/home/prelude.ts/node_modules/list/dist/index.js:1347:28)
    at Object.takeWhile (/home/emmanuel/home/prelude.ts/node_modules/list/dist/index.js:1373:12)
    at Vector.takeWhile (/home/emmanuel/home/prelude.ts/dist/src/Vector.js:565:29)
    at Vector.span (/home/emmanuel/home/prelude.ts/dist/src/Vector.js:590:26)
    at Object.linesByIndentGetTagContents (/home/emmanuel/home/prelude.ts/dist/scripts/make_doc_extra/helpers.js:34:20)
    at putStaticMethodsOnTop (/home/emmanuel/home/prelude.ts/dist/scripts/make_doc_extra/classes.js:15:22)
    at r (/home/emmanuel/home/prelude.ts/dist/src/Function.js:202:49)
    at /home/emmanuel/home/prelude.ts/dist/src/Function.js:209:91
```

**Comment by emmanueltouzery** [comment]
confirmed still present in 2.0.15

**Comment by paldepind** [comment]
This was fixed along with the issues in #67 as far as I'm aware.
